src/main/java/lotto/controller/LottoController.java

```diff
@@ -0,0 +1,82 @@
+package lotto.controller;
+
+
+import lotto.view.InputView;
+import lotto.view.OutputView;
+import lotto.model.*;
+
+public class LottoController {
+
+    public void start() {
+        LottoAmount amount = inputAmount();
+        LottoTickets lottoNum = buyLotto(amount);
+        WinningNumber winningNum = inputWinningNumber();
+        BonusNumber bonusNum = inputBonusNumber(winningNum);
+        WinningResult winningResult = calculateWinningResult(lottoNum, winningNum, bonusNum);
+        calculateEarningRate(winningResult, amount);
+    }
+
+    private LottoAmount inputAmount() {
+        try {
+            String LottoAmount = InputView.inputLottoAmount();
+            return new LottoAmount(LottoAmount);
+        } catch (IllegalArgumentException e) {
+            OutputView.printErrorMessage(e.getMessage());
+            return inputAmount();
+        }
+    }
+
+    private LottoTickets buyLotto(LottoAmount amount) {
+        LottoTickets lottoTickets = new LottoTickets(amount.getLottoCount());
+        OutputView.printLottoCount(amount.getLottoCount());
+        OutputView.printLottoTickets(lottoTickets);
+        return lottoTickets;
+    }
+
+    private WinningNumber inputWinningNumber() {
+        try {
+            String winningNum = InputView.inputWinningLottoNumbers();
+            return new WinningNumber(winningNum);
+        } catch (IllegalArgumentException e) {
+            OutputView.printErrorMessage(e.getMessage());
+            return inputWinningNumber();
+        }
+    }
+
+    private BonusNumber inputBonusNumber(WinningNumber winningNum) {
+        try {
+            String bonusNum = InputView.inputBonusNumber();
+            BonusNumber bonusNumber = new BonusNumber(bonusNum);
+            validateDuplicatedNumber(winningNum, bonusNumber);
+            return bonusNumber;
+        } catch (IllegalArgumentException e) {
+            OutputView.printErrorMessage(e.getMessage());
+            return inputBonusNumber(winningNum);
+        }
+    }
+
+    private void validateDuplicatedNumber(WinningNumber winningNum, BonusNumber bonusNumber) {
+        if (winningNum.getWinningNumbers().contains(bonusNumber.getBonusNumber())) {
+            throw new IllegalArgumentException("보너스 번호는 당첨 번호와 중복될 수 없습니다.");
+        }
+    }
+
+    private WinningResult calculateWinningResult(LottoTickets tickets, WinningNumber winningNum, BonusNumber bonusNumber) {
+        WinningResult winningResult = new WinningResult();
+        for (Lotto lotto : tickets.getTickets()) {
+            int matchCount = lotto.getMatchCount(winningNum);
+            boolean hasBonus = lotto.hasBonus(bonusNumber);
+            LottoRanking ranking = LottoRanking.determineRanking(matchCount, hasBonus);
+            winningResult.addRanking(ranking);
+        }
```

Comment:
객체의 결합도를 줄인 방식이 인상적이네요. 이건 저도 참고해야겠어요.

Reply:
결합도에 신경쓰지 않는다면, 아래와 같이 코드를 간결하게 만들 수 있어요.

우선 lotto의 정보를 이용해서 ranking을 생성하는 책임을 lotto에 넘길 수 있어요.
```suggestion
        WinningResult winningResult = new WinningResult();
        for (Lotto lotto : tickets.getTickets()) {
            LottoRanking ranking = lotto.getRanking(winningNum, bonusNumber);
            winningResult.addRanking(ranking);
        }
```

그리고 winningResult를 만드는 책임을 winningResult에게 넘길 수 있어요.
```suggestion
        WinningResult winningResult = new WinningResult(tickets, winningNum, bonusNumber);
```

이런 방법도 있다는 걸 알아두시면 좋을 것 같아요 ㅎㅎ

Reply:
좋은 피드백 감사합니다